### PR TITLE
checking for nothing in deploy

### LINKED ISSUE #870: Error when opening PR
At the end of a failed deploy():

```julia
Would you like to actually open the pull request or just prepare it?
 > Go ahead, open it
   No, just prepare it and let me look at it first
ERROR: MethodError: no method matching length(::Nothing)
Closest candidates are:
  length(::Base.AsyncGenerator) at asyncmap.jl:410
  length(::LibGit2.GitBlob) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LibGit2/src/blob.jl:3
  length(::Base.EnvDict) at env.jl:132
  ...
Stacktrace:
 [1] _zip_min_length(::Tuple{Nothing}) at ./iterators.jl:312
 [2] _zip_min_length(::Tuple{Array{Symbol,1},Nothing}) at ./iterators.jl:308 (repeats 2 times)
 [3] length(::Base.Iterators.Zip{Tuple{Array{String,1},Array{Symbol,1},Nothing}}) at ./iterators.jl:302
 [4] _similar_for(::UnitRange{Int64}, ::Type{Tuple{String,Symbol,Any}}, ::Base.Iterators.Zip{Tuple{Array{String,1},Array{Symbol,1},Nothing}}, ::Base.HasLength) at ./array.jl:597
 [5] _collect(::UnitRange{Int64}, ::Base.Iterators.Zip{Tuple{Array{String,1},Array{Symbol,1},Nothing}}, ::Base.HasEltype, ::Base.HasLength) at ./array.jl:630
 [6] collect(::Base.Iterators.Zip{Tuple{Array{String,1},Array{Symbol,1},Nothing}}) at ./array.jl:624
 [7] print_build_tarballs(::Base.GenericIOBuffer{Array{UInt8,1}}, ::BinaryBuilder.WizardState) at /home/mbesancon/.julia/packages/BinaryBuilder/9oqch/src/wizard/deploy.jl:31
 [8] yggdrasil_deploy(::BinaryBuilder.WizardState, ::Bool) at /home/mbesancon/.julia/packages/BinaryBuilder/9oqch/src/wizard/deploy.jl:123
 [9] _deploy(::BinaryBuilder.WizardState) at /home/mbesancon/.julia/packages/BinaryBuilder/9oqch/src/wizard/deploy.jl:241
 [10] deploy(::BinaryBuilder.WizardState) at /home/mbesancon/.julia/packages/BinaryBuilder/9oqch/src/wizard/deploy.jl:280
 [11] deploy() at /home/mbesancon/.julia/packages/BinaryBuilder/9oqch/src/wizard/deploy.jl:270
 [12] top-level scope at REPL[3]:1
```

This is also the case when trying the other options (writing to local file and stdout)

### PR DESCRIPTION
**Pull request by matbesancon**
Closes #870

### CODE REVIEW

File: src/wizard/deploy.jl [hunk 1]
```diff
@@ -25,7 +25,7 @@ function print_build_tarballs(io::IO, state::WizardState)
         """
     end
 
-    if state.files === nothing
+    if state.files === nothing || state.file_varnames === nothing
         products_string = "Product[\n]"
```
Comment (giordano) [inline src/wizard/deploy.jl:29]: I'd check that the zip below isn't nothing

Comment (matbesancon) [reply src/wizard/deploy.jl:29]: zip itself does not return nothing:
```
julia> zip([], nothing)
zip(Any[], nothing)

julia> collect(zip([], nothing))
ERROR: MethodError: no method matching length(::Nothing)
```

Comment (giordano) [reply src/wizard/deploy.jl:29]: :disappointed:

Comment (matbesancon) [reply src/wizard/deploy.jl:29]: you mean the `state.files`?

Comment (giordano) [reply src/wizard/deploy.jl:29]: My worry is that `state.file_kinds` can be `nothing` and we again don't catch it

Comment (matbesancon) [reply src/wizard/deploy.jl:29]: done

Comment (giordano) [reply src/wizard/deploy.jl:29]: Done what? :thinking:

Comment (matbesancon) [reply src/wizard/deploy.jl:29]: well, it was done then. Now it is pushed :see_no_evil:

File: src/wizard/deploy.jl [hunk 2]
```diff
@@ -25,7 +25,7 @@ function print_build_tarballs(io::IO, state::WizardState)
         """
     end
 
-    if state.files === nothing
+    if state.files === nothing || state.file_kinds === nothing || state.file_varnames === nothing
```
Comment (giordano) [inline src/wizard/deploy.jl:28]: This is a bit shorter:
```suggestion
    if any(isnothing, (state.files, state.file_kinds, state.file_varnames))
```

File: src/wizard/deploy.jl (hunk 1 above)
Comment (giordano) [reply src/wizard/deploy.jl:29]: Just realised I commented on the wrong line :weary:

**Pull request merged**